src/http/conn.rs

```diff
@@ -155,7 +155,8 @@ where I: Io,
     }
 
     fn maybe_park_read(&mut self) {
-        if self.io.poll_read().is_ready() {
+        // TODO: this needs fixing
```

Comment:
Hrm, so as the comment below tries to describe, this situation would occur when hyper has read the end of a body, such as described by a `Content-Length` header, and then stops, as it's waiting on the response side to determine if it should read the next request.

If the socket were to already have some of the bytes in its buffer, then our not reading to end would mean `TcpStream` would not unpark the task again to read more. That's a scenario when hyper needs to park, and then unpark later when the response finishes and keeps the connection alive.

So, to get the same as poll_read from now on, does that mean we need to store our own `read_has_blocked` flag?